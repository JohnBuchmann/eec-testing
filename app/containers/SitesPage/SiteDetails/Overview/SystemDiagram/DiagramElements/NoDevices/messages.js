/*
 * System diagram auto sync dc bus messages
 *
 * This contains all the text for the auto sync dc bus.
 */
import { defineMessages } from 'react-intl';

export const scope =
  'app.containers.SitesPage.SiteDetails.Overview.SystemDiagram.NoDevices';

export default defineMessages({
  noDevicesTitle: {
    id: `${scope}.noDevicesTitle`,
    defaultMessage: 'Site Configuration Unavailable',
  },
});
