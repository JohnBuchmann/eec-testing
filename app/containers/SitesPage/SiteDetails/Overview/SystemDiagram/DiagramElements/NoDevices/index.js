import React from 'react';
import PropTypes from 'prop-types';
import { injectIntl } from 'react-intl';
import { makeStyles } from '@material-ui/core/styles';
import { Colors } from 'Theme';
import messages from './messages';

/**
 * Declare UI styles
 */
const useStyles = makeStyles({
  noDevicesTitle: {
    fontFamily: 'Inter, sans-serif',
    fontSize: '12px',
    fontWeight: '300',
    lineHeight: 'normal',
    letterSpacing: '.6px',
    color: Colors.white,
  },
});

/**
 * NoDevices
 * Display no devices message
 */
function NoDevices(props) {
  const { formatMessage } = props.intl;

  const classes = useStyles();
  return (
    <>
      <rect
        stroke={Colors.white}
        fill={Colors.nandor}
        x={158}
        y={267}
        width={855}
        height={22}
        rx={4}
      />
      <text x={480} y={283} className={classes.noDevicesTitle}>
        {formatMessage(messages.noDevicesTitle)}
      </text>
    </>
  );
}

NoDevices.propTypes = {
  intl: PropTypes.any.isRequired,
};

export default injectIntl(NoDevices);
