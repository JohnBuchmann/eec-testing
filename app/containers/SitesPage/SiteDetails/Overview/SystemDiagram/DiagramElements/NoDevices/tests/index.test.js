import { mount } from 'enzyme';
import React from 'react';
import { IntlProvider } from 'react-intl';
import NoDevices from '..';

describe('NoDevices', () => {
  it('should display NoDevices', () => {
    const actualComponent = mount(
      <svg xmlns="http://www.w3.org/2000/svg">
        <IntlProvider locale="en">
          <NoDevices />
        </IntlProvider>
      </svg>
    );

    expect(actualComponent).toMatchSnapshot();
  });
});
