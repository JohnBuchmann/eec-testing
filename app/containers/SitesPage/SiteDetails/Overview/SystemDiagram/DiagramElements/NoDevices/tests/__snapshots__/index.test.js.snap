// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoDevices should display NoDevices 1`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
>
  <IntlProvider
    locale="en"
  >
    <InjectIntl(NoDevices)>
      <NoDevices
        intl={
          Object {
            "defaultFormats": Object {},
            "defaultLocale": "en",
            "formatDate": [Function],
            "formatHTMLMessage": [Function],
            "formatMessage": [Function],
            "formatNumber": [Function],
            "formatPlural": [Function],
            "formatRelative": [Function],
            "formatTime": [Function],
            "formats": Object {},
            "formatters": Object {
              "getDateTimeFormat": [Function],
              "getMessageFormat": [Function],
              "getNumberFormat": [Function],
              "getPluralFormat": [Function],
              "getRelativeFormat": [Function],
            },
            "locale": "en",
            "messages": Object {},
            "now": [Function],
            "onError": [Function],
            "textComponent": "span",
            "timeZone": null,
          }
        }
      >
        <rect
          fill="#515e51"
          height={22}
          rx={4}
          stroke="#ffffff"
          width={855}
          x={158}
          y={267}
        />
        <text
          className="makeStyles-noDevicesTitle-1"
          x={480}
          y={283}
        >
          Site Configuration Unavailable
        </text>
      </NoDevices>
    </InjectIntl(NoDevices)>
  </IntlProvider>
</svg>
`;
